### PR TITLE
Promote arc-serialization-persistence logic to master behind `serial` url param

### PR DESCRIPTION
**Pull request by sjmiles**
Include `serial` as a url param to shell apps to turn on Arc persistence via _runtime serialization_ instead of _steps_.

### CODE REVIEW

File: runtime/arc.js [hunk 1]
```diff
@@ -234,7 +234,11 @@ ${this.activeRecipe.toString()}`;
       context
     });
     // TODO: pass tags through too
-    manifest.stores.forEach(store => arc._registerStore(store, []));
+    manifest.stores.forEach(store => {
+      if (store.constructor.name == 'StorageStub')
```
Comment (mmandlis) [inline runtime/arc.js]: Is this for backward compatibility? When would the `store` not be a `StorageStub`?
probably question for @shans

**Comment by sjmiles** [comment]
Yes, I pulled PRs into this branch along the way, so the commit list is weird.

But since the 'files changed' list looks to be correct, I didn't worry about it.

File: runtime/arc.js (hunk 1 above)
Comment (sjmiles) [reply runtime/arc.js]: Yes, this comes from @shans #1259.

File: runtime/manifest.js [hunk 2]
```diff
@@ -30,6 +30,20 @@ class ManifestError extends Error {
...
+  constructor(type, id, name, storageKey, storageProviderFactory) {
+    this.type = type;
+    this.id = id;
+    this.name = name;
+    this.storageKey = storageKey;
+    this.storageProviderFactory;
+  }
+
+  inflate() {
+    return this.storageProviderFactory.connect(this.id, this.type, this.storageKey);
```
Comment (shaper) [inline runtime/manifest.js]: I filed https://github.com/PolymerLabs/arcs/issues/1442 to track adding unit tests for this and the logic above which is fine to do separately.

**Pull request merged**